dds/DCPS/XTypes/DynamicDataBase.cpp

```diff
@@ -346,9 +346,93 @@ DDS::ReturnCode_t DynamicDataBase::unsupported_method(const char* method_name, b
 }
 
 #ifndef OPENDDS_NO_CONTENT_SUBSCRIPTION_PROFILE
-DDS::ReturnCode_t DynamicDataBase::get_simple_value(DCPS::Value& /*value*/, DDS::MemberId /*id*/)
+namespace {
+  template <typename T>
+  DDS::ReturnCode_t get_some_value(DCPS::Value& value, DDS::MemberId id, DDS::DynamicData& dyn,
+                                   DDS::ReturnCode_t (DDS::DynamicData::* pmf)(T&, DDS::MemberId))
+  {
+    T v;
+    const DDS::ReturnCode_t ret = (dyn.*pmf)(v, id);
+    if (ret == DDS::RETCODE_OK) {
+      value = v;
+    }
+    return ret;
+  }
+
+  DDS::ReturnCode_t get_some_value(DCPS::Value& value, DDS::MemberId id, DDS::DynamicData& dyn,
+                                   DDS::ReturnCode_t (DDS::DynamicData::* pmf)(char*&, DDS::MemberId))
+  {
+    CORBA::String_var v;
+    const DDS::ReturnCode_t ret = (dyn.*pmf)(v, id);
+    if (ret == DDS::RETCODE_OK) {
+      value = v.in();
+    }
+    return ret;
+  }
+
+  DDS::ReturnCode_t get_some_value(DCPS::Value& value, DDS::MemberId id, DDS::DynamicData& dyn,
+                                   DDS::ReturnCode_t (DDS::DynamicData::* pmf)(ACE_CDR::WChar*&, DDS::MemberId))
+  {
+    CORBA::WString_var v;
+    const DDS::ReturnCode_t ret = (dyn.*pmf)(v, id);
+    if (ret == DDS::RETCODE_OK) {
+      value = v.in();
+    }
+    return ret;
+  }
+}
+
+DDS::ReturnCode_t DynamicDataBase::get_simple_value(DCPS::Value& value, DDS::MemberId id)
 {
-  return unsupported_method("DynamicDataBase::get_simple_value");
+  DDS::DynamicTypeMember_var dtm;
+  if (type_->get_member(dtm, id) != DDS::RETCODE_OK) {
+    return DDS::RETCODE_ERROR;
+  }
+  DDS::MemberDescriptor_var md;
+  if (dtm->get_descriptor(md) != DDS::RETCODE_OK) {
+    return DDS::RETCODE_ERROR;
+  }
+  DDS::DynamicType_var member_type = get_base_type(md->type());
+  const TypeKind member_kind = member_type->get_kind();
+  switch (member_kind) {
+  case TK_BOOLEAN:
+    return get_some_value(value, id, *this, &DDS::DynamicData::get_boolean_value);
+  case TK_BYTE:
+    return get_some_value(value, id, *this, &DDS::DynamicData::get_byte_value);
+  case TK_INT8:
+    return get_some_value(value, id, *this, &DDS::DynamicData::get_int8_value);
+  case TK_INT16:
+    return get_some_value(value, id, *this, &DDS::DynamicData::get_int16_value);
+  case TK_INT32:
+    return get_some_value(value, id, *this, &DDS::DynamicData::get_int32_value);
+  case TK_INT64:
+    return get_some_value(value, id, *this, &DDS::DynamicData::get_int64_value);
+  case TK_UINT8:
+    return get_some_value(value, id, *this, &DDS::DynamicData::get_uint8_value);
+  case TK_UINT16:
+    return get_some_value(value, id, *this, &DDS::DynamicData::get_uint16_value);
+  case TK_UINT32:
+    return get_some_value(value, id, *this, &DDS::DynamicData::get_uint32_value);
+  case TK_UINT64:
+    return get_some_value(value, id, *this, &DDS::DynamicData::get_uint64_value);
```

Comment:
I might be wrong about this, but because `Value` just takes either a int64 or uint64, I think you could combine the signed and unsigned cases into two cases and the casting in `get_int64_value` and `get_uint64_value` will take care of the rest.

```suggestion
  case TK_INT8:
  case TK_INT16:
  case TK_INT32:
  case TK_INT64:
    return get_some_value(value, id, *this, &DDS::DynamicData::get_int64_value);
  case TK_UINT8:
  case TK_UINT16:
  case TK_UINT32:
  case TK_UINT64:
    return get_some_value(value, id, *this, &DDS::DynamicData::get_uint64_value);
```

Reply:
I don't think this is correct: "Value just takes either a int64 or uint64"

Reply:
I meant as far as integer values as concerned.

Reply:
Integer types of different sizes are supported - see constructors of `DCPS::Value`